### PR TITLE
Feature/120 gcp svc act multi proj

### PR DESCRIPTION
**Pull request by rtomlinson-latacora**
Extending this featuring a little more.
Add --all CLI arg to explicitly allow for scanning of all projects a service account or user has access to.
Change `else` clause in gcp/provider.py to raise Exception if unable to determine default project id or project id/folder id/organization id is not provided on CLI.
README.md documentation

### CODE REVIEW

**Comment by j4v** [comment]
Has this branch merged in https://github.com/nccgroup/ScoutSuite/pull/151? I'm asking as the CLI parser implementation is changing, so would make sense to take this into account before adding new parameters.

File: ScoutSuite/__main__.py [hunk 1]
```diff
@@ -97,6 +97,7 @@ def main(passed_args=None):
                                                     key_file=args.key_file,
                                                     user_account=args.user_account,
                                                     service_account=args.service_account,
+                                                    all=args.all,
```
Comment (j4v) [inline ScoutSuite/__main__.py]: let's use something more obvious such as `gcp_scan_all`

in truth would be good to prepend all provider-specific arguments with `provider_`

File: ScoutSuite/cli_parser.py [hunk 2]
```diff
@@ -197,13 +197,19 @@ def __init__(self, default_args=None):
 
         self.parser.add_argument('--project-id',
                                  action='store',
+                                 required='--all' not in sys.argv,
```
Comment (j4v) [inline ScoutSuite/cli_parser.py]: I believe this is the "old" parser, which is being refactored in https://github.com/nccgroup/ScoutSuite/pull/151

File: ScoutSuite/providers/gcp/provider.py [hunk 3]
```diff
@@ -45,7 +45,7 @@ def __init__(self, project_id=None, folder_id=None, organization_id=None,
 
         super(GCPProvider, self).__init__(report_dir, timestamp, services, skipped_services, thread_config)
 
-    def authenticate(self, key_file=None, user_account=None, service_account=None, **kargs):
+    def authenticate(self, key_file=None, user_account=None, service_account=None, all=False, **kargs):
```
Comment (j4v) [inline ScoutSuite/providers/gcp/provider.py]: I don't think this is right, `all` should be available in `self`, as for `project_id`/`folder_id`/etc

File: ScoutSuite/__main__.py (hunk 1 above)
Comment (rtomlinson-latacora) [reply ScoutSuite/__main__.py]: Changed flag to gcp_scan_all

File: ScoutSuite/providers/gcp/provider.py (hunk 3 above)
Comment (rtomlinson-latacora) [reply ScoutSuite/providers/gcp/provider.py]: Moved to GCPProvider object and made `gcp-scan-all` available as kwargs

**Comment by rtomlinson-latacora** [comment]
I merged #151 CLI Refactoring into this branch just now. The latest push should have all the changes associated with the cli refactor.

**Comment by Aboisier** [comment]
Hey, thanks for your contribution again, I'll wait until #151 is merged to review this though.

**Comment by zer0x64** [comment]
#151 Has been mergd!

**Comment by j4v** [comment]
Working on this, should be able to merge today.

**Comment by j4v** [comment]
Closing this PR as changes are pushed through https://github.com/nccgroup/ScoutSuite/pull/169.

**Pull request closed without merging**